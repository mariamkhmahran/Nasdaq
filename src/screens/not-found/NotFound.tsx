import React from 'react';
import { useNavigate } from 'react-router';
import { motion } from 'framer-motion';

import LogoN from 'components/logo-n';
import { PATHS } from '../../globals';

import { Container, FlatBtton, FourOhFour, Text, Word } from './styles';

export const NotFound: React.FC = () => {
  const navigate = useNavigate();

  return (
    <motion.div
      initial={{ opacity: 0, height: '100vh' }}
      animate={{ opacity: 1 }}
      transition={{ type: 'keyframes' }}
    >
      <Container>
        <FourOhFour>404</FourOhFour>
        <Text>
          <Word>Page</Word>
          <Word>
            <LogoN />
            ot
          </Word>
          <Word>Found</Word>
        </Text>
        <FlatBtton onClick={() => navigate(PATHS.exploreScreen)}>
          Explore Stocks <i className="fas fa-chevron-right" />
        </FlatBtton>
      </Container>
    </motion.div>
  );
};
