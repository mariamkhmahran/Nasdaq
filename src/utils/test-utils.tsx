import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'overmind-react';
import { createOvermind } from 'overmind';
import { config } from '../overmind-state';
import Theme from 'assets/styles/Theme';

const overmind = createOvermind(config, { devtools: true });

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <Provider value={overmind}>
      <ThemeProvider theme={Theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </Provider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
