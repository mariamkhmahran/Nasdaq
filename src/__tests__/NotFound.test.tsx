import { waitFor, render } from 'utils/test-utils';
import { NotFound } from 'screens/not-found/NotFound';

describe('404 Screen', () => {
  test('Renders 404 page correctly', async () => {
    const container = render(<NotFound />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
