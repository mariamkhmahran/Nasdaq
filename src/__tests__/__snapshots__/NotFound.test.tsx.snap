// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Screen Renders 404 page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="opacity: 0; height: 100vh;"
      >
        <div
          class="sc-bdvvtL cKGMAp"
        >
          <div
            class="sc-gsDKAQ cCJDUu"
          >
            404
          </div>
          <div
            class="sc-dkPtRN fEhLwK"
          >
            <div
              class="sc-hKwDye eSuRMW"
            >
              Page
            </div>
            <div
              class="sc-hKwDye eSuRMW"
            >
              <svg
                data-name="logosandtypes com"
                id="logosandtypes_com"
                viewBox="0 0 150 150"
                width="40px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M112.74,12.85h34.07L115.16,100a6.06,6.06,0,0,1-5.71,4H76v0a6.07,6.07,0,0,0,5.09-4Z"
                  fill="rgba(255, 255, 255, 0)"
                  pathLength="1"
                  stroke="#fff"
                  stroke-dasharray="0px 1px"
                  stroke-dashoffset="0px"
                  stroke-width="2"
                />
                <path
                  d="M73.28,102.7A5.73,5.73,0,0,1,68,99.23L61.19,80.47l11.61-32a5.77,5.77,0,0,1,10.73-.31l6.79,18.68-11.61,32A5.73,5.73,0,0,1,73.28,102.7Z"
                  fill="rgba(255, 255, 255, 0)"
                  pathLength="1"
                  stroke="#fff"
                  stroke-dasharray="0px 1px"
                  stroke-dashoffset="0px"
                  stroke-width="2"
                />
                <path
                  d="M75.75,43.33v0a6,6,0,0,0-5.45,4L38.65,134.5H4.58L36.23,47.33a6.07,6.07,0,0,1,5.7-4Z"
                  fill="rgba(255, 255, 255, 0)"
                  pathLength="1"
                  stroke="#fff"
                  stroke-dasharray="0px 1px"
                  stroke-dashoffset="0px"
                  stroke-width="2"
                />
              </svg>
              ot
            </div>
            <div
              class="sc-hKwDye eSuRMW"
            >
              Found
            </div>
          </div>
          <div
            class="sc-eCImPb bRQzrg"
          >
            Explore Stocks 
            <i
              class="fas fa-chevron-right"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="opacity: 0; height: 100vh;"
    >
      <div
        class="sc-bdvvtL cKGMAp"
      >
        <div
          class="sc-gsDKAQ cCJDUu"
        >
          404
        </div>
        <div
          class="sc-dkPtRN fEhLwK"
        >
          <div
            class="sc-hKwDye eSuRMW"
          >
            Page
          </div>
          <div
            class="sc-hKwDye eSuRMW"
          >
            <svg
              data-name="logosandtypes com"
              id="logosandtypes_com"
              viewBox="0 0 150 150"
              width="40px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M112.74,12.85h34.07L115.16,100a6.06,6.06,0,0,1-5.71,4H76v0a6.07,6.07,0,0,0,5.09-4Z"
                fill="rgba(255, 255, 255, 0)"
                pathLength="1"
                stroke="#fff"
                stroke-dasharray="0px 1px"
                stroke-dashoffset="0px"
                stroke-width="2"
              />
              <path
                d="M73.28,102.7A5.73,5.73,0,0,1,68,99.23L61.19,80.47l11.61-32a5.77,5.77,0,0,1,10.73-.31l6.79,18.68-11.61,32A5.73,5.73,0,0,1,73.28,102.7Z"
                fill="rgba(255, 255, 255, 0)"
                pathLength="1"
                stroke="#fff"
                stroke-dasharray="0px 1px"
                stroke-dashoffset="0px"
                stroke-width="2"
              />
              <path
                d="M75.75,43.33v0a6,6,0,0,0-5.45,4L38.65,134.5H4.58L36.23,47.33a6.07,6.07,0,0,1,5.7-4Z"
                fill="rgba(255, 255, 255, 0)"
                pathLength="1"
                stroke="#fff"
                stroke-dasharray="0px 1px"
                stroke-dashoffset="0px"
                stroke-width="2"
              />
            </svg>
            ot
          </div>
          <div
            class="sc-hKwDye eSuRMW"
          >
            Found
          </div>
        </div>
        <div
          class="sc-eCImPb bRQzrg"
        >
          Explore Stocks 
          <i
            class="fas fa-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
